ctapipe/reco/sklearn.py

```diff
@@ -0,0 +1,628 @@
+
+        valid = self.quality_query.get_table_mask(table)
+        energy[valid], is_valid[valid] = self._predict(key, table[valid])
+
+        result = Table(
+            {
+                f"{self.model_cls}_tel_energy": energy,
+                f"{self.model_cls}_tel_is_valid": is_valid,
+            }
+        )
+        add_defaults_and_meta(
+            result,
+            ReconstructedEnergyContainer,
+            prefix=self.model_cls,
+            stereo=False,
+        )
+        return result
+
+
+class ParticleClassifier(SKLearnClassficationReconstructor):
+    """
+    Predict dl2 particle classification
+    """
+
+    #: Name of the target table column for training
+    target = "true_shower_primary_id"
+    property = "classification"
+
+    positive_class = Integer(
+        default_value=0,
+        help="Particle id (in simtel system) of the positive class. Default is 0 for gammas.",
+    ).tag(config=True)
+
+    def __call__(self, event: ArrayEventContainer) -> None:
+        for tel_id in event.trigger.tels_with_trigger:
+            table = collect_features(event, tel_id, self.instrument_table)
+            table = self.feature_generator(table)
+            passes_quality_checks = self.quality_query.get_table_mask(table)[0]
+
+            if passes_quality_checks:
+                prediction, valid = self._predict_score(
+                    self.subarray.tel[tel_id],
+                    table,
+                )
+
+                container = ParticleClassificationContainer(
+                    prediction=prediction[0],
+                    is_valid=valid[0],
+                )
+            else:
+                container = ParticleClassificationContainer(
+                    prediction=np.nan, is_valid=False
+                )
+
+            container.prefix = f"{self.model_cls}_tel"
+            event.dl2.tel[tel_id].classification[self.model_cls] = container
+
+        self.stereo_combiner(event)
+
+    def predict_table(self, key, table: Table) -> Table:
+        """Predict on a table of events"""
+        table = self.feature_generator(table)
+
+        n_rows = len(table)
+        score = np.full(n_rows, np.nan)
+        is_valid = np.full(n_rows, False)
+
+        valid = self.quality_query.get_table_mask(table)
+        score[valid], is_valid[valid] = self._predict_score(key, table[valid])
+
+        result = Table(
+            {
+                f"{self.model_cls}_tel_prediction": score,
+                f"{self.model_cls}_tel_is_valid": is_valid,
+            }
+        )
+        add_defaults_and_meta(
+            result, ParticleClassificationContainer, prefix=self.model_cls, stereo=False
+        )
+        return result
+
+
+class CrossValidator(Component):
+    """Class to train sklearn based reconstructors in a cross validation"""
+
+    n_cross_validations = Int(5).tag(config=True)
+
+    output_path = Path(
+        default_value=None,
+        allow_none=True,
+        directory_ok=False,
```

Comment:
please add a help string here.  It should at least explain format is the output in. Is is a PDF plot or a picked class? Impossible to tell.  I assumed pkl like the other output, but then looking at the file, it seems to be hdf5. That's not obvious.